### PR TITLE
Get rid of the dependency between TestTools and TestHelpers

### PR DESCRIPTION
**Pull request by testableapple**
### 🔗 Issue Links

Resolve https://github.com/GetStream/ios-issues-tracking/issues/644

### 🎯 Goal

- To improve developer experience, move required stuff from TestHelpers to TestTools

### 📝 Summary

- Just moved stuff required by Unit tests from TestHelpers to TestTools

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

gif

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -27,6 +27,10 @@ let package = Package(
             targets: ["StreamChatTestMockServer"]
         ),
     ],
+    dependencies: [
```
Comment (martinmitrevski) [inline Package.swift:30]: Why not embed these in the test tools / mock server?

**Review by martinmitrevski — COMMENTED** [review]
I think we can also address the Difference/Swifter dependency here.
Apart from that, looks good. I assume customers won't have any build errors while updating to this, right?

File: Package.swift (hunk 1 above)
Comment (testableapple) [reply Package.swift:30]: Great idea, let's try that. This will increase the PR grammatically, but that will be just embedded libs.

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

**Pull request merged**